src/Interpreters/Cache/QueryCache.cpp

```diff
@@ -186,17 +202,15 @@ String queryStringFromAST(ASTPtr ast)
 
 }
 
-/// Hashing of ASTs must consider aliases (issue #56258)
-static constexpr bool ignore_aliases = false;
-
 QueryCache::Key::Key(
     ASTPtr ast_,
+    String current_database,
```

Comment:
Should we also hash the settings used? The results and header might change depending on the settings used (in the query, the session, etc).

Reply:
Yes, that's a good idea.

But, allow me to do that separately as a follow-up. I want to downport the current PR as it fixes a security issue. Unhashed settings "only" pose a correctness issue.

Reply:
https://github.com/ClickHouse/ClickHouse/pull/64205